### PR TITLE
try to use std::make_unique instead of new for unique_ptr's

### PR DESCRIPTION
**Pull request by boyceg**
Part of the war on `new`.

### IBAMR Pull Request Checklist
- [ ] Does the test suite pass?
- [x] Was clang-format (i.e., `make indent`) run? For more information see
      `scripts/formatting/README.md`.
- [x] Were relevant issues cited? Please remember to add `Fixes #12345` to close
      the issue automatically if we are fixing the problem.
- [x] Is this a change others will want to know about? If so, then has a
      changelog entry been added?
- [x] If new data structures have been added to a class then have they been set
      up appropriately for restarts? If so, ensure that the restart version
      number is incremented.
- [x] Does this change include a bug fix or new functionality? If so, a new test
      or tests should be added. New tests should run quickly (less than a minute
      in release mode). If possible, an older test should gain a new option so
      that we do not need to compile more test executables.
- [x] Did you (if your account has permission to do so) set relevant labels on
      GitHub for the pull request?

### CODE REVIEW

**Comment by boyceg** [comment]
Ugh, I forgot that we don't yet require C++14. The year is 2024; let's update. See: #1739.

**Comment by boyceg** [comment]
Anyone wanna review this one?

**Review by abarret — APPROVED** [review]
We never ran the modernize scripts on the examples, so there's more things we can update such as `auto`. But that can be something for later.

**Comment by drwells** [comment]
You'll need to rebase this.

**Comment by boyceg** [comment]
@drwells: rebased. Shall I merge once I get 👍 on the testsuite?

File: examples/IBFE/explicit/ex9/example.cpp [hunk 1]
```diff
@@ -283,7 +283,7 @@ main(int argc, char* argv[])
         EquationSystems* bndry_equation_systems = ib_method_ops->getFEDataManager()->getEquationSystems();
 
         // Setup solid systems.
-        std::unique_ptr<EquationSystems> solid_equation_systems(new EquationSystems(solid_mesh));
+        std::unique_ptr<EquationSystems> solid_equation_systems = std::make_unique<EquationSystems>(solid_mesh);
```
Comment (drwells) [inline examples/IBFE/explicit/ex9/example.cpp:286]:
```suggestion
        auto solid_equation_systems = std::make_unique<EquationSystems>(solid_mesh);
```
We can't do this with ternary ifs but we could do it it here.

**Review by drwells — APPROVED** [review]

File: examples/IBFE/explicit/ex9/example.cpp (hunk 1 above)
Comment (boyceg) [reply examples/IBFE/explicit/ex9/example.cpp:286]: Ah my regex did not find those.

**Pull request merged**